plot/plot.app.js

```diff
@@ -1157,16 +1289,128 @@
+                                                        }
+                                                    }
+                                                    
+                                                    
+                                                    */
+
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("graphic_link_pattern")){
+                                                        var idx="";
+                                                        switch(data.points[0].data.type){
+                                                            case "pie":
+                                                                idx=data.points[0].data.labels.indexOf(data.points[0].label.toString());
+                                                                break;
+                                                            case "bar":
+                                                                if(data.points[0].data.orientation=="h")
+                                                                    idx=data.points[0].data.y.indexOf(data.points[0].y.toString());
+                                                                else
+                                                                    idx=data.points[0].data.x.indexOf(data.points[0].x.toString());
+                                                                break;
+                                                        }
+                                                        if( data.points[0].data.graphic_link_pattern[idx]!=false && data.points[0].data.graphic_link_pattern[idx]!=undefined){
+                                                            var url=data.points[0].data.graphic_link_pattern[idx];
+                                                            window.open(url,'_blank');
+                                                        }
+                                                    }
+                                                }),
+                                                element[0].on('plotly_legendclick', function(data){
+                                                    //This event occurs on click of legend
+                                                    if(data.hasOwnProperty("data")){
+                                                            if(data.data[0].hasOwnProperty("legend_clickable_links") ){
+                                                                var idx=data.data[0].labels.indexOf(data.label.toString());
+                                                                if(data.data[0].legend_clickable_links[idx]!=false && data.data[0].legend_clickable_links[idx]!=undefined){
+                                                                    window.open(data.data[0].legend_clickable_links[0],'_blank');
+                                                                    return false;
```

Comment:
We need to do this check here as well before returning false:
```
if(data.layout.hasOwnProperty("disable_default_legend_click") && data.layout.disable_default_legend_click==true)
```

Reply:
Made relevant changes